src/sysdir.c

```diff
@@ -82,15 +82,22 @@ static int git_sysdir_guess_global_dirs(git_buf *out)
 #else
 	int error;
 	uid_t uid, euid;
+	const char *sandbox_id;
 
 	uid = getuid();
 	euid = geteuid();
 
+	/*
+	 * If APP_SANDBOX_CONTAINER_ID is set, we are running in a
+	 * sandboxed environment on macOS.
+	 */
+	sandbox_id = getenv("APP_SANDBOX_CONTAINER_ID");
```

Comment:
This needs to be `p_getenv`

Reply:
Can't find `p_getenv` anywhere p_getenv: `Implicit declaration of function 'p_getenv'`. Is this defined somewhere externally?

Reply:
`p_getenv` is also not used anywhere else in the project, except a non active code branch in `remote.c` which is not compiled.

Reply:
I *think* he meant `git__getenv()`, which is used to query the `HOME` environment variable.

Reply:
Yes, but `git__getenv` is just a wrapper around `getenv` that returns a buffer, which we don't need. So what would be the point of that?

Reply:
`p_` is the POSIX-compat prefix. This could be a suggestion to implement `p_getenv` on macOS with a fallback for `$HOME` (which could be considered — otherwise there's a big chance the next `getenv("HOME")` added will suffer the same issue). But that felt heavy-handed, and I'm missing the point made as well…

Reply:
Don't implement a `p_getenv` - we used to have one, that's why we added `git__getenv` because there's no POSIXy way to do this on Windows.

I think this is :+1: as-is, since it's non-Windows only.

Reply:
Yeah, sorry about that. A quick grep for `getenv` returned `p_getenv` to me, which is why I assumed that we had the POSIX wrappers in place. One reason why I hate ifdef'fed blocks, I'll create a PR to just remove that block.